src/types/platforms-types/multi-track-multiprint-contract.ts

```diff
@@ -83,13 +73,19 @@ const mapArtistProfile = ({ apiTrack, nft, contract }: { apiTrack: any, nft?: NF
   }
 };
 
-const mapNFTtoTrackID = (nft: NFT): string => {
-  const trackName = getTrait(nft, 'Track');
-  return `ethereum/${formatAddress(nft.contractAddress)}/${trackName}`;
+const mapNFTtoTrackID = (nft: NFT, extractor?: Extractor): string => {
+  if (!extractor) {
+    throw new Error('No extractor provided');
+  }
+  const id = slugify(extractor(nft));
+  if (!id) {
+    throw new Error('ID not extracted correctly');
+  }
+  return `ethereum/${formatAddress(nft.contractAddress)}/${id}`;
```

Comment:
I think we should aim to make the idExtractor responsible for the whole id, not just the suffix, so let's implement that for this first case.